### PR TITLE
RLN-16: update to work with Gradle 7

### PR DESCRIPTION
**Pull request by arndey**
[RLN-16](https://soramitsu.atlassian.net/browse/RLN-16)

### CODE REVIEW

File: src/main/groovy/jp/co/soramitsu/devops/SoraPlugin.groovy [hunk 1]
```diff
@@ -106,9 +106,25 @@ class SoraPlugin implements Plugin<Project> {
...
+
+        var filteredDependsOn = allDependsOn.findAll { dTask ->
+            for (String toRemove in tasksToRemove) {
+                if (dTask == toRemove) {
+                    return false
+                }
+            }
+            return true
+        }
+        task.dependsOn.addAll(filteredDependsOn)
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/SoraPlugin.groovy:128]:
```suggestion
        var filteredDependsOn = allDependsOn.findAll { dTask -> !tasksToRemove.contains(dTask)}
        task.dependsOn.addAll(filteredDependsOn)
```

File: src/test/groovy/jp/co/soramitsu/devops/tasks/TasksTest.groovy [hunk 2]
```diff
@@ -34,9 +34,9 @@ class TasksTest extends Specification {
         hasTask(result, SoraTask.dockerPush)
         hasTask(result, SoraTask.dockerVersion)
 
-        and: "has no this tasks"
-        !hasTask(result, 'jacocoTestReport')
-        !hasTask(result, 'jacocoTestCoverageVerification')
+//        and: "has no this tasks"
+//        !hasTask(result, 'jacocoTestReport')  TODO
```
Comment (Mingela) [inline src/test/groovy/jp/co/soramitsu/devops/tasks/TasksTest.groovy:38]: created another task for future, please refer to that in the todo:
`TODO: RLN-46`

**Review by Mingela — APPROVED** [review]
please consider the minor change, also wait till reviewed by others

**Review by AntonKtrue — APPROVED** [review]

**Review by BulatSaif — APPROVED** [review]

**Comment by BulatSaif** [comment]
we need to replace Travis-ci with github actions

File: projects/01-java-app/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-bin.zip
```
Comment (Mingela) [inline projects/01-java-app/gradle/wrapper/gradle-wrapper.properties:3]: maybe let's have some of the projects continue to have an old Gradle version to check backward compatibility?

File: .github/workflows/github-actions.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+name: test-deploy-gradle-sora-plugin
+on: [push]
```
Comment (ravenolf) [inline .github/workflows/github-actions.yml:2]:
```suggestion
on: [pull_request]
```

File: .github/workflows/github-actions.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+          chmod +x gradlew
+          chmod +x publish.sh
+      - name: Run Gradle Wrapper Daemon, Build, then Test
+        run: |
+          ./gradlew --daemon
+          ./gradlew build -x test
+          ./gradlew test
+      - name: Run Deployment Script
+        run: ./publish.sh
+        shell: bash
```
Comment (ravenolf) [inline .github/workflows/github-actions.yml:28]:
```suggestion
      - name: Run Deployment Script
        run: ./gradlew publishPlugins
        shell: bash
```
Let's move it to another workflow with:
```yaml
on:
  push:
    branches: ["master"]
```

Comment (ravenolf) [reply .github/workflows/github-actions.yml:28]: Also you can delete publish since it was just wrapper with travis token

**Review by ravenolf — DISMISSED** [review]

File: .github/workflows/github-actions.yml (hunk 5 above)
Comment (Mingela) [reply .github/workflows/github-actions.yml]: didn't see your comments, just was plying with that, partially fixed without knowing the warnings

File: .github/workflows/github-actions.yml (hunk 4 above)
Comment (Mingela) [reply .github/workflows/github-actions.yml:2]: I guess it's on `push` to the `master` (latest) or explicit release versions corresponding to GH tags, we should add value setting for the version responsible env var

**Comment by Mingela** [comment]
Is Travis check still required? If yes, please remove that

**Review by Mingela — APPROVED** [review]
please merge via 'squash' option

**Review by BulatSaif — APPROVED** [review]

**Pull request merged**